text/0001-cdk-update.md

```diff
@@ -0,0 +1,300 @@
+---
+rfc pr: [#335](https://github.com/aws/aws-cdk-rfcs/pull/335)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/1
+---
+
+# CDK Update
+
+Shorten the development iteration speed -- the "edit-compile-test" loop -- for
+CDK applications.
+
+## Working Backwards
+
+### CHANGELOG
+
+- feat(cli): implement `cdk update` and `cdk update --watch` for asset updates
+
+### README
+
+The `cdk update` command accelerates the edit-compile-test loop for your CDK
+application during development by inspecting the assets and stack resources in
+your application, identifying those that can be updated in-place without a full
+CloudFormation stack udpate, and doing so using AWS service APIs directly. It
+can either do this as an interactive command or by running continuously and
+monitoring the input files.
+
+For supported construct types (see list, below) the update command will identify
+assets whose content has changed since the publication of the CDK, assemble them
+from sources if necessary, and then publish them to their asset storage medium.
+It will then use AWS SDK commands to directly modify the CDK resources to use
+the new asset paths. The update command acts on one or more stacks in the
+application, and will only perform the shortcut if all changes are updateable.
+
+A stack is "updateable" if the stack is explicitly marked as updateable, and it
+contains any assets supported by the update process. At this time, only Lambda
+and ECS images are supported, and so any stack with one of those is updateable.
```

Comment:
This paragraph is not what we want, I think - the support is based on the types of resources, not on types of Assets.

I would re-word it to be something like:

```suggestion
A stack is "updateable" if the stack is explicitly marked as updateable, and it
contains resources supported by the update process. At this time, only Lambda
Functions and ECS Services are supported, however more resource types will be added to this list in the future.
```